Objects/dict-common.h

```diff
@@ -8,37 +8,35 @@ typedef struct {
     PyObject *me_value; /* This field is only meaningful for combined tables */
 } PyDictKeyEntry;
 
-/* dict_lookup_func() returns index of entry which can be used like DK_ENTRIES(dk)[index].
+/* _Py_dict_lookup() returns index of entry which can be used like DK_ENTRIES(dk)[index].
  * -1 when no entry found, -3 when compare raises error.
  */
-typedef Py_ssize_t (*dict_lookup_func)
-    (PyDictObject *mp, PyObject *key, Py_hash_t hash, PyObject **value_addr);
+Py_ssize_t _Py_dict_lookup(PyDictObject *mp, PyObject *key, Py_hash_t hash, PyObject **value_addr);
+
 
 #define DKIX_EMPTY (-1)
 #define DKIX_DUMMY (-2)  /* Used internally */
 #define DKIX_ERROR (-3)
 
+typedef enum {
+    DICT_KEYS_GENERAL = 0,
+    DICT_KEYS_UNICODE = 1,
+    DICT_KEYS_UNICODE_NO_DUMMY = 2,
```

Comment:
`_Py_dict_lookup()` don't have specialized case for DICT_KEYS_UNICODE_NO_DUMMY. Can we delete this kind?

This kind is used only here.
https://github.com/python/cpython/blob/a316f34dd8dceb34269e176a10b0fbc8dea55980/Objects/dictobject.c#L1226

But it can be replaced with `mp->ma_used > mp->ma_keys->dk_nentries`.